--- fhist-1.21.D001/common/output.h	2012-07-13 14:47:15.000000000 +1000
+++ fhist-1.21.D001.NEW/common/output.h	2015-06-02 13:05:43.000000000 +1000
@@ -72,7 +72,7 @@
 
 #ifndef DEBUG
 #ifdef __GNUC__
-
+#if 0
 extern __inline const char *output_filename(output_ty *fp) { return
         fp->vptr->filename(fp); }
 extern __inline long output_ftell(output_ty *fp) { return
@@ -85,7 +85,7 @@
         fp->vptr->write(fp, data, len); }
 extern __inline void output_flush(output_ty *fp) {
         fp->vptr->flush(fp); }
-
+#endif
 #else /* !__GNUC__ */
 
 #define output_filename(fp) ((fp)->vptr->filename(fp))
--- fhist-1.21.D001/common/input.h	2012-07-13 14:47:15.000000000 +1000
+++ fhist-1.21.D001.NEW/common/input.h	2015-06-02 13:05:09.000000000 +1000
@@ -65,6 +65,7 @@
 struct string_ty *input_one_line(input_ty *);
 
 #ifdef __GNUC__
+#if 0
 extern __inline long input_read(input_ty *fp, void *data, long len)
         { if (len <= 0) return 0; if (fp->pushback_len > 0) {
         fp->pushback_len--; *(char *)data = fp->pushback_buf[
@@ -78,6 +79,7 @@
         { return fp->vptr->length(fp); }
 extern __inline long input_ftell(input_ty *fp)
         { return fp->vptr->ftell(fp) - fp->pushback_len; }
+#endif
 #else /* !__GNUC__ */
 #ifndef DEBUG
 #define input_name(fp) ((fp)->vptr->name(fp))
--- fhist-1.21.D001/common/config.messy.h	2012-07-13 14:47:15.000000000 +1000
+++ fhist-1.21.D001.NEW/common/config.messy.h	2015-06-02 13:03:23.000000000 +1000
@@ -78,6 +78,9 @@
 #ifndef _BSD_SOURCE
 #define _BSD_SOURCE
 #endif
+#ifndef _DEFAULT_SOURCE
+#define _DEFAULT_SOURCE
+#endif
 #endif
 
 #endif /* COMMON_CONFIG_MESSY_H */
