--- SparsePOP303/defaultParameter.m	2018-09-18 13:19:23.000000000 +1000
+++ SparsePOP-master/defaultParameter.m	2020-12-22 13:06:44.000000000 +1100
@@ -246,7 +246,7 @@
 if ~isfield(param,'relaxOrder')
     param.relaxOrder = 1; 
 %   param.relaxOrder will be updated to
-%       max{the minimum relaxation order ?½omega_{?½max}, param.relaxOrder}.
+%       max{the minimum relaxation order ? omega_{?max}, param.relaxOrder}.
 end
 
 if ~isfield(param,'sparseSW')
@@ -461,9 +461,17 @@
 
 if ~isfield(param,'symbolicMath')
 	%   Default:
-	A = ver('Symbolic');
+  if exist('OCTAVE_VERSION','builtin')
+    pkg_id = 'symbolic';
+    pkg_name = pkg_id;
+  else
+    pkg_id = 'Symbolic';
+    pkg_name = 'Symbolic Math Toolbox';
+  end
+  
+  A = ver(pkg_id);
 	if ~isempty(A)
-		x = strfind(A.Name, 'Symbolic Math Toolbox');
+	x = strfind(A.Name, pkg_name);
 		if ~isempty(x)
 			param.symbolicMath = 1;
 		else
--- SparsePOP303/compileSparsePOP.m	2018-09-11 11:35:46.000000000 +1000
+++ SparsePOP-master/compileSparsePOP.m	2020-12-22 13:06:44.000000000 +1100
@@ -20,33 +20,15 @@
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %
 
-if exist('verLessThan') ~= 2
-	error('Get verLessThan.m');
-end
-
-
-if verLessThan('matlab', '7.3') 
-	MexFlags = ' -O -Dlinux=0 ';
-elseif strcmp(computer, 'GLNXA64') || strcmp(computer, 'MACI64')
-	MexFlags = ' -O -Dlinux=1 -largeArrayDims ';
-elseif strcmp(computer, 'GLNX86')  || strcmp(computer, 'MACI')
-	MexFlags = ' -O -Dlinux=0 ';
-elseif strcmp(computer, 'PCWIN64') 
-	MexFlags = ' -O -Dlinux=1 -largeArrayDims ';
-elseif strcmp(computer, 'PXWIN') 
-	MexFlags = ' -O -Dlinux=0 ';
-else 
-	MexFlags = ' -O -Dlinux=0 ';
-end
-
-LIBfiles = ' conversion.cpp spvec.cpp polynomials.cpp sup.cpp clique.cpp mysdp.cpp Parameters.cpp ';
-if ispc % Windows family create .obj files
-        OBJfiles = strrep(LIBfiles,'.cpp','.obj');
-else
+MexFlags = ' -O2 -Dlinux=1 -DMATLAB_MEX_FILE -DOCTAVE ';
+LIBfiles = strcat(' conversion.cpp spvec.cpp polynomials.cpp sup.cpp ', ...
+                  ' clique.cpp mysdp.cpp Parameters.cpp ');
         OBJfiles = strrep(LIBfiles,'.cpp','.o');
-end
 
-eval('cd subPrograms/Mex');
+mpwd=pwd;
+mpath=mfilename('fullpath');
+mpath=mpath(1:strchr(mpath,filesep,1,'last'));
+cd(strcat(mpath,filesep,'subPrograms',filesep,'Mex'));
 fprintf('Compiling Libraries...');
 command = ['mex -c ' MexFlags LIBfiles];
 eval(command);
@@ -59,54 +41,6 @@
 command = ['mex ' MexFlags ' mexconv2.cpp ' OBJfiles ];
 eval(command);
 fprintf('done\n');
-eval('cd ../../');
-
-%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-% mex files of SparseCoLO
-%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-
-if verLessThan('matlab', '7.3') 
-	MexFlags = ' -O -Dlinux=0 ';
-elseif strcmp(computer, 'GLNXA64') || strcmp(computer, 'MACI64')  
-	MexFlags = ' -O -Dlinux=1 -largeArrayDims ';
-elseif strcmp(computer, 'GLNX86')  || strcmp(computer, 'MACI')
-	MexFlags = ' -O -Dlinux=0 ';
-elseif strcmp(computer, 'PCWIN64')
-	MexFlags = ' -O -Dlinux=1 -largeArrayDims ';
-elseif strcmp(computer, 'PCWIN')
-	MexFlags = ' -O -Dlinux=0 ';
-else
-	MexFlags = ' -O -Dlinux=0 ';
-end
-
-LIBfiles = [' ccputime.cpp'];
-if ispc % Windows family create .obj files
-        OBJfiles = strrep(LIBfiles,'.cpp','.obj');
-else
-        OBJfiles = strrep(LIBfiles,'.cpp','.o');
-end
-
-eval('cd V260SubPrograms/SparseCoLO/mex'); 
-fprintf('Compiling Libraries...');
-command = ['mex -c ' MexFlags LIBfiles];
-eval(command);
-fprintf('done\n');
-
-clear mexFiles
-
-mexFiles{1} = 'mexForestConvert.cpp';
-mexFiles{2} = 'mexMaxSpanningTree2.cpp';
-mexFiles{3} = 'mexPrimalOneSDP2.cpp';
-% mexFiles{4} = 'mexArrowTriDQOP.cpp';
-% mexFiles{5} = 'mexDiagTriDQOP.cpp';
-for i=1:length(mexFiles)
-    mexFileName = mexFiles{i};
-    fprintf('Compiling %s...',mexFileName);
-    command = ['mex ' MexFlags mexFileName OBJfiles];
-    eval(command);
-    fprintf('done\n');
-end
-
-eval('cd ../../../');
+cd(mpwd);
 
 fprintf('Compilation finished successfully.\n');
--- SparsePOP303/V210SubPrograms/BroydenTriLS.m	2018-09-11 11:35:47.000000000 +1000
+++ SparsePOP-master/V210SubPrograms/BroydenTriLS.m	2020-12-22 13:06:44.000000000 +1100
@@ -5,9 +5,9 @@
 %   minimize   \sum_{i=1}^n objPoly{i}^2
 %   subject to inequPolySys = []; x_1 >= 0
 % Here
-%   objPoly{1} = (3|2x_1)x_1|2x_2+1
-%   objPoly{i} = (3|2x_i)x_i|x_{i|1}| 2x_{i+1}+1 (i=2,...,n-1)
-%   objPoly{n} = (3|2x_n)x_n|x_{n|1}+1
+%   objPoly{1} = (3-2x_1)x_1-2x_2+1
+%   objPoly{i} = (3-2x_i)x_i-x_{i-1}- 2x_{i+1}+1 (i=2,...,n-1)
+%   objPoly{n} = (3-2x_n)x_n-x_{n-1}+1
 % 
 % 
 % [] = solveBroydenTri(nDim);
--- SparsePOP303/readGMS.m	2018-09-18 14:41:11.000000000 +1000
+++ SparsePOP-master/readGMS.m	2020-12-22 13:06:44.000000000 +1100
@@ -189,8 +189,7 @@
 eqOrIneq = cell(1,noOfEquations);
 rightValue = cell(1,noOfEquations);
 for i=1:noOfEquations
-    idx = findstr(listOfEquations{i},'=');
-    %equationNames{i}
+    idx = strfind(listOfEquations{i},'=');
     if isempty(idx) || length(idx) ~= 2
         error('## The constaint of ''%s'' should have ''=E='', ''=G='' or ''=L=''.\n## Should check the kind of the constriant and/or the position of '';''. ', equationNames{i});
         %elseif length(idx) > 2
@@ -211,7 +210,7 @@
 p = 1;
 temp = [];
 while (p <= noOfTerms) && isempty(temp)
-    temp = findstr(listOfTerms{objRow}{p},'objvar');
+    temp = strfind(listOfTerms{objRow}{p},'objvar');
     if ~isempty(temp)
         objTerm = p;
     end
@@ -354,9 +353,9 @@
 if eqTo2ineqSW == 0
     for i=1:noOfEquations
         pointer = i;
-        [statusSW, poly, msg] = convToPolynomial(noOfVariables,varNames,...
+        [statusSW, Poly, msg] = convToPolynomial(noOfVariables,varNames,...
             listOfTerms{i},eqOrIneq{i},rightValue{i});
-        ineqPolySys{i} = poly;
+        ineqPolySys{i} = Poly;
         if statusSW ~= 0
             error('%s## Should check the %d%s constraint.', msg, i, thWord(i));
         end
@@ -372,23 +371,23 @@
     for i=1:noOfEquations
         if (eqOrIneq{i} == 'G')
             pointer = pointer + 1;
-            [statusSW, poly, msg] = convToPolynomial(noOfVariables,varNames,...
+            [statusSW, Poly, msg] = convToPolynomial(noOfVariables,varNames,...
                 listOfTerms{i},eqOrIneq{i},rightValue{i});
-            ineqPolySys{pointer} = poly;
+            ineqPolySys{pointer} = Poly;
         elseif (eqOrIneq{i} == 'L')
             pointer = pointer + 1;
-            [statusSW, poly, msg] = convToPolynomial(noOfVariables,varNames,...
+            [statusSW, Poly, msg] = convToPolynomial(noOfVariables,varNames,...
                 listOfTerms{i},eqOrIneq{i},rightValue{i});
-            ineqPolySys{pointer} = poly;
+            ineqPolySys{pointer} = Poly;
         else
             pointer = pointer + 1;
-            [statusSW, poly, msg] = convToPolynomial(noOfVariables,varNames,...
+            [statusSW, Poly, msg] = convToPolynomial(noOfVariables,varNames,...
                 listOfTerms{i},'G',rightValue{i});
-            ineqPolySys{pointer} = poly;
+            ineqPolySys{pointer} = Poly;
             pointer = pointer + 1;
-            [statusSW, poly, msg] = convToPolynomial(noOfVariables,varNames,...
+            [statusSW, Poly, msg] = convToPolynomial(noOfVariables,varNames,...
                 listOfTerms{i},'L',rightValue{i}+eqTolerance);
-            ineqPolySys{pointer} = poly;
+            ineqPolySys{pointer} = Poly;
         end
         if statusSW ~= 0
             error('%s## Should check the %d%s constraint.', msg, i, thWord(i));
@@ -407,8 +406,8 @@
 	tmpL{2} = strcat('-1*',binVarNames{i});
 	tmpE = 'E';
 	tmpR = 0;	
-        [statusSW, poly, msg] = convToPolynomial(noOfVariables,varNames,tmpL,tmpE,tmpR);
-        ineqPolySys{i+noOfEquations} = poly;
+        [statusSW, Poly, msg] = convToPolynomial(noOfVariables,varNames,tmpL,tmpE,tmpR);
+        ineqPolySys{i+noOfEquations} = Poly;
         if statusSW ~= 0
             error('%s## Should check the %d%s constraint.', msg, i, thWord(i+noOfEquations));
         end
@@ -434,7 +433,7 @@
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 function allStatements = fromFile(fileName)
 fileIDX = fopen(fileName, 'r');
-allStatements = [];
+allStatements = '';
 nextSW = 0;
 while 1
     oneLine = fgetl(fileIDX);
@@ -460,7 +459,7 @@
 return
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 function [nextSW, statements] = fromOneLine(oneLine, nextSW)
-statements = [];
+statements = '';
 if nextSW == 0
     oneLine = deblank(oneLine);
     oneLine = strtrim(oneLine);
@@ -502,7 +501,7 @@
             nextSW = 0;
            return 
         end
-        idx = findstr(oneLine, ';');
+        idx = strfind(oneLine, ';');
     end
     if ~isempty(oneLine)
         statements = [statements, oneLine];
@@ -540,13 +539,13 @@
 return
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 function allStatements = removeWhiteSpaces(fileName, allStatements)
-idx = findstr(allStatements, ';');
+idx = strfind(allStatements, ';');
 if isempty(idx)
     error('## ''%s'' does not have any '';''.', fileName);
 elseif idx(end) ~= length(allStatements)
     error('## '';'' does not exist at the end of the last statement of ''%s''.', fileName);
 end
-NewAllStatements = [];
+NewAllStatements = '';
 sidx = 1;
 for i=1:length(idx)
     eidx = idx(i);
@@ -571,7 +570,7 @@
 return
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 function [varNames, allStatements] = getVarName(fileName, allStatements)
-idx = findstr(allStatements, ';');
+idx = strfind(allStatements, ';');
 if isempty(idx)
     error('## ''%s'' does not have any '';''.', fileName);
 elseif idx(end) ~= length(allStatements)
@@ -621,7 +620,7 @@
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 function [posVarNames, allStatements] = getPosvarName(fileName, allStatements)
 % get the definition of Positive Variables
-idx = findstr(allStatements, ';');
+idx = strfind(allStatements, ';');
 sidx = 1;
 posVarNames = [];
 for i=1:length(idx)
@@ -679,7 +678,7 @@
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 function [binVarNames, allStatements] = getBinaryName(fileName, allStatements)
 % get the definition of Binary Variables
-idx = findstr(allStatements, ';');
+idx = strfind(allStatements, ';');
 sidx = 1;
 binVarNames = [];
 for i=1:length(idx)
@@ -742,7 +741,7 @@
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 function [equationNames, allStatements] = getEquationName(fileName, allStatements)
 % get the definition of Equations
-idx = findstr(allStatements, ';');
+idx = strfind(allStatements, ';');
 if isempty(idx)
     error('## ''%s'' does not have the line of ''Equations''.',fileName); 
 elseif idx(end) ~= length(allStatements)
@@ -778,7 +777,7 @@
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 function [NewEquationNames, listOfEquations, allStatements] = getEquation(fileName, allStatements, equationNames, symbolicMath)
 % get the objective function and all the constraints
-idx = findstr(allStatements, ';');
+idx = strfind(allStatements, ';');
 if isempty(idx)
     error('## ''%s'' does not have the line of ''Equations''.',fileName); 
 elseif idx(end) ~= length(allStatements)
@@ -845,7 +844,7 @@
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 function [lbd, ubd, fixed, allStatements] = getLowerUpper(fileName, allStatements, varNames)
 % get all upper and lower bounds.
-idx = findstr(allStatements, ';');
+idx = strfind(allStatements, ';');
 sidx = 1;
 noOfVariables = size(varNames,2);
 lbd = -1.0e10* ones(1,noOfVariables);
@@ -938,6 +937,9 @@
 		%	loca = loca(1) -2; % for objvar
 		%end
 		%oneLine(1:loca)
+      if exist('OCTAVE_VERSION','builtin');
+        tempf = collect(sym(oneLine(1:loca)), 'objvar');
+      else
 		%20180911 H. Waki modified
 		if verLessThan('matlab','9.4')
 			tempf = collect(sym(oneLine(1:loca)), 'objvar');
@@ -945,6 +947,7 @@
 			syms objvar
 			tempf = collect(str2sym(oneLine(1:loca)), objvar);
 		end
+      end
 		oneLinetmp = char(vpa(expand(tempf),20));
 		oneLine = strcat(oneLinetmp, oneLine(loca+1:end));
 	end
@@ -1003,7 +1006,7 @@
 function objRow = getObeRow(listOfEquations, noOfEquations)
 objRow = [];
 for i=1:noOfEquations
-    temp = findstr(listOfEquations{i},'objvar');
+    temp = strfind(listOfEquations{i},'objvar');
     if isempty(temp) ~= 1
         objRow = [objRow,i];
     end
@@ -1065,7 +1068,7 @@
     else
         k = k+1;
         ii = 1;
-        SignVec = [];
+        SignVec = '';
         while 1
             if (formerPart(1) == '-') || (formerPart(1) == '+')
                 SignVec(ii) = formerPart(1);
@@ -1079,7 +1082,7 @@
                 break;
             end
         end
-        minus_num = length(findstr(SignVec, '-'));
+        minus_num = length(strfind(SignVec, '-'));
         if ~isempty(SignVec) && mod(minus_num, 2) == 1
             listOfTerms{k} = '-';
         elseif ~isempty(SignVec) && mod(minus_num, 2) == 0
@@ -1093,7 +1096,7 @@
         ll = length(formerPart);
         lenOneTerm = length(oneTerm);
         if (lenOneTerm > 0) && ((oneTerm(lenOneTerm) == 'e') || (oneTerm(lenOneTerm) == 'E')) ...
-                && findstr(oneTerm,'.') && (ll > 0) && isempty(findstr(oneTerm,'*'))
+                && strfind(oneTerm,'.') && (ll > 0) && isempty(strfind(oneTerm,'*'))
             signE = formerPart(1);
             if (signE == '+') || (signE == '-')
                 [oneTerm1,formerPart] = strtok(formerPart,'-+;');
@@ -1172,7 +1175,7 @@
 while isempty(oneTerm) ~= 1
     [oneVariable,oneTerm] = strtok(oneTerm,'*');
     kk = length(oneVariable);
-    pp = findstr(oneVariable,'^');
+    pp = strfind(oneVariable,'^');
     powerPart = 1;
     if isempty(pp) ~= 1
         powerPart = str2num(oneVariable(pp+1:kk));
@@ -1196,27 +1199,27 @@
 end
 return
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-function [statusSW, poly, msg] = convToPolynomial(noOfVariables,varNames,listOfTerms,...
+function [statusSW, Poly, msg] = convToPolynomial(noOfVariables,varNames,listOfTerms,...
     eqOrIneq,rightValue)
 noOfTerms = size(listOfTerms,2);
 statusSW = 0;
 msg = [];
 if strcmp(eqOrIneq, 'E') || strcmp(eqOrIneq, 'e')
-    poly.typeCone = -1;
+    Poly.typeCone = -1;
 else
-    poly.typeCone = 1;
+    Poly.typeCone = 1;
 end
-poly.sizeCone = 1;
-poly.degree = 0;
-poly.dimVar = noOfVariables;
+Poly.sizeCone = 1;
+Poly.degree = 0;
+Poly.dimVar = noOfVariables;
 if abs(rightValue) > 1.0e-10
-    poly.noTerms = noOfTerms + 1;
-    poly.supports = sparse(1,poly.dimVar);
-    poly.coef = -rightValue;
+    Poly.noTerms = noOfTerms + 1;
+    Poly.supports = sparse(1,Poly.dimVar);
+    Poly.coef = -rightValue;
 else
-    poly.noTerms = noOfTerms;
-    poly.supports = [];
-    poly.coef = [];
+    Poly.noTerms = noOfTerms;
+    Poly.supports = [];
+    Poly.coef = [];
 end
 for p=1:noOfTerms
     oneTerm = listOfTerms{p};
@@ -1224,33 +1227,33 @@
     if statusSW ~= 0
        return 
     end
-    poly.supports = [poly.supports; supportVec];
-    poly.coef = [poly.coef;coef];
+    Poly.supports = [Poly.supports; supportVec];
+    Poly.coef = [Poly.coef;coef];
     %	full(supportVec)
     degree = full(sum(supportVec));
-    poly.degree = max(poly.degree,degree);
+    Poly.degree = max(Poly.degree,degree);
 end
-% poly.degree
+% Poly.degree
 if eqOrIneq == 'L' || eqOrIneq == 'l'
-    poly.coef = - poly.coef;
+    Poly.coef = - Poly.coef;
 end
 
 return
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 function [lbd]  = posToInEqPolySys(noOfVariables,varNames,oneVariable,lbd)
 
-%poly.typeCone = 1;
-%poly.sizeCone = 1;
-%poly.degree = 1;
-%poly.dimVar = noOfVariables;
-%poly.noTerms = 1;
-%poly.supports = sparse(zeros(1,poly.dimVar));
-%poly.coef = [1];
+%Poly.typeCone = 1;
+%Poly.sizeCone = 1;
+%Poly.degree = 1;
+%Poly.dimVar = noOfVariables;
+%Poly.noTerms = 1;
+%Poly.supports = sparse(zeros(1,Poly.dimVar));
+%Poly.coef = [1];
 
 i = 1;
 while (i <= noOfVariables)    
     if strcmp(oneVariable,varNames{i})
-        %		poly.supports(1,i) = 1;
+        %		Poly.supports(1,i) = 1;
         lbd(i) = max(lbd(i),0);
         break;
         %i = noOfVariables + 1;
--- SparsePOP303/example/POPformat/randomwithEQ.m	2018-09-11 11:35:47.000000000 +1000
+++ SparsePOP-master/example/POPformat/randomwithEQ.m	2020-12-22 13:06:44.000000000 +1100
@@ -409,7 +409,7 @@
     same = 0; 
     i=1;
     sparseZeros = sparse(zeros(1,nDim-startIndex+1)); 
-    while (same==0) & (i <= noOfCliques)
+    while (same==0) && (i <= noOfCliques)
       diff = oneClique(1,startIndex:nDim) - cliqueSet(i,startIndex:nDim); 
       diff = max(diff,sparseZeros); 
       if length(find(diff)) == 0
--- SparsePOP303/example/POPformat/randomUnconst.m	2018-09-11 11:35:47.000000000 +1000
+++ SparsePOP-master/example/POPformat/randomUnconst.m	2020-12-22 13:06:44.000000000 +1100
@@ -379,7 +379,7 @@
     same = 0; 
     i=1;
     sparseZeros = sparse(zeros(1,nDim-startIndex+1)); 
-    while (same==0) & (i <= noOfCliques)
+    while (same==0) && (i <= noOfCliques)
       diff = oneClique(1,startIndex:nDim) - cliqueSet(i,startIndex:nDim); 
       diff = max(diff,sparseZeros); 
       if length(find(diff)) == 0
--- SparsePOP303/example/POPformat/randomConst.m	2018-09-11 11:35:47.000000000 +1000
+++ SparsePOP-master/example/POPformat/randomConst.m	2020-12-22 13:06:44.000000000 +1100
@@ -380,7 +380,7 @@
     same = 0; 
     i=1;
     sparseZeros = sparse(zeros(1,nDim-startIndex+1)); 
-    while (same==0) & (i <= noOfCliques)
+    while (same==0) && (i <= noOfCliques)
       diff = oneClique(1,startIndex:nDim) - cliqueSet(i,startIndex:nDim); 
       diff = max(diff,sparseZeros); 
       if length(find(diff)) == 0
--- SparsePOP303/example/POPformat/genPIS.m	2018-09-11 11:35:47.000000000 +1000
+++ SparsePOP-master/example/POPformat/genPIS.m	2020-12-22 13:06:44.000000000 +1100
@@ -23,24 +23,30 @@
 % 2011-06-22 H.Waki
 %
 
+u = 10; % given integer
+k = ceil(2*nDim/3);
+if nDim - k < 1
+  error('nDim should be more than 1.');
+end
+
+if exist('OCTAVE_VERSION','builtin')
+  rand('state', 3201);
+  rvec = rand(nDim, 1);
+else
 mv= ver('matlab');
 mv = str2num(mv.Version);
-
 if mv > 7.7
 	s = RandStream('mt19937ar','Seed', 3201);
 else
 	rand('twister',3201);
 end
-u = 10; % given integer
-k = ceil(2*nDim/3);
-if nDim - k < 1
-	error('nDim should be more than 1.');
-end
 if mv > 7.7
 	rvec = rand(s, nDim, 1);
 else
 	rvec = rand(nDim, 1);
 end
+end
+
 for i=1:k
 	a(i, 1) = ceil(u*rvec(i)); 
 end
--- SparsePOP303/solveExample.m	2018-09-24 15:59:49.000000000 +1000
+++ SparsePOP-master/solveExample.m	2020-12-22 13:06:44.000000000 +1100
@@ -189,11 +189,7 @@
 param0.printLevel = [2,2];
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 if nargin == 0
-    %    if strcmp(param0.SDPsolver,'sdpa')
-    %        probNumbers = [1:12,14:72, 74:85];
-    %    else
-    probNumbers = [1:72 ,74:81, 83:95];
-    %    end
+    probNumbers = [1:95];
     if param0.symbolicMath == 0
         if ~isfield(param0, 'SDPsolver') || strcmp(param0.SDPsolver,'sedumi')
             probNumbers = [1:12, 14:34, 37, 43:44, 47:53, 55:62, 64:69, 74:81, 83:87, 89:90, 93:95];
@@ -201,19 +197,6 @@
             probNumbers = [1:12, 14:34, 37, 43:44, 47:53, 55:62, 64:69, 74:87, 89:90, 93:95];
         end
     end
-else
-    if ~isfield(param0, 'SDPsolver') || strcmp(param0.SDPsolver,'sedumi')
-        %
-        % 2010-01-08 H. Waki
-        %
-        % SeDuMi_1_21 can not solve SDP relaxation problems
-        % generated from qp3.gms and optControl2(200).
-        % So, we skip these problems if 'probNumbers' contains them.
-        idx73 = find(probNumbers == 73);
-        idx82 = find(probNumbers == 82);
-        idx = [idx73, idx82];
-        probNumbers(idx) = [];
-    end
 end
 % for my macbook
 %idx6 = find(probNumbers == 6);
--- SparsePOP303/subPrograms/Mex/conversion.cpp	2018-09-11 11:35:47.000000000 +1000
+++ SparsePOP-master/subPrograms/Mex/conversion.cpp	2020-12-22 13:06:44.000000000 +1100
@@ -2041,6 +2041,9 @@
                     + bassinfo_st[infolist[k]].sup.vap_size * polyinfo_st[infolist[k]].sup.pnz_size;
             numele += polyinfo_st[infolist[k]].sup.pnz_size * bassinfo_st[infolist[k]].sup.pnz_size;
             k++;
+            if(k == stsize){
+              break;
+            }
 		cVec = InfoTable[k];
         }
         int bpsize, bvsize;
@@ -2073,6 +2076,9 @@
 		minkovsum(polyinfo_st[infolist[k]].sup, bassinfo_st[infolist[k]].sup, minsups);
 		pushsups(minsups, allsups);
 		k++;
+          if(k == stsize){
+            break;
+          }
 		cVec = InfoTable[k];
 	}
         
--- SparsePOP303/subPrograms/Mfiles/removeDependColumns1.m	2018-09-11 11:35:47.000000000 +1000
+++ SparsePOP-master/subPrograms/Mfiles/removeDependColumns1.m	2020-12-22 13:06:44.000000000 +1100
@@ -152,7 +152,9 @@
             % columns
             toBecheckedColumns = toBecheckedColumns(nzColumns);
             % Apply the LU factorization
-			%
+			if exist('OCTAVE_VERSION','builtin')
+              [L,U,PVect,Q] = lu(U, 'vector');
+            else
 			% 2012-01-11 H.Waki
 			% lu(mat, 'vector') is not implemented in 7.3 or earlier. 
 			if exist('verLessThan') ~= 2
@@ -166,6 +168,7 @@
 			else
 				[L,U,PVect] = lu(U,'vector');
 			end
+            end
             %->
             if debugSW == 1
                 fprintf('Apply the LU factorization to update the candidate column indices of linearly independent columns; U = \n');
--- SparsePOP303/subPrograms/Mfiles/my_unique.m	2018-09-11 11:35:47.000000000 +1000
+++ SparsePOP-master/subPrograms/Mfiles/my_unique.m	2020-12-22 13:06:44.000000000 +1100
@@ -1,7 +1,12 @@
 function [C, ia, ic] = my_unique(A, msg)
+
 if ~strcmp(msg, 'rows');
 	error('msg should be rows in my_unique.');
 end
+
+if exist('OCTAVE_VERSION','builtin')
+  [C, ia, ic] = unique(A, msg);
+else
 if verLessThan('matlab', '8.0.1')
 	% This part is the same as unique in R2012b or earlier version
 	[C, ia, ic] = unique(A, msg);
@@ -10,4 +15,6 @@
 	% We use legacy mode of unique.
 	[C, ia, ic] = unique(A, msg, 'last', 'legacy');
 end
+end
+
 return
--- SparsePOP303/subPrograms/Mfiles/convert2.m	2018-09-11 11:35:47.000000000 +1000
+++ SparsePOP-master/subPrograms/Mfiles/convert2.m	2020-12-22 13:06:44.000000000 +1100
@@ -123,7 +123,13 @@
         for i=1:objPoly0.noTerms          
             pp = objPoly0.supports(i,j); 
             if pp >= 1 
+              if exist('OCTAVE_VERSION','builtin')
+                cc = sym2poly((sym(full(trans.Amat(j,j)))*y ...
+                               + sym(full(trans.bVect(j,1)))) ...
+                              ^sym(full(pp)))';
+              else
                 cc = sym2poly((trans.Amat(j,j)*y+trans.bVect(j,1))^pp)';
+              end
                 supSet = objPoly0.supports(i*ones(pp+1,1),:);
                 supSet(:,j) = (pp:-1:0)';
 %                
@@ -177,7 +183,13 @@
                 for i=1:ineqPolySys0{k}.noTerms
                     pp = ineqPolySys0{k}.supports(i,j); 
                     if pp >= 1 
+                      if exist('OCTAVE_VERSION','builtin')
+                        cc = sym2poly((sym(full(trans.Amat(j,j)))*y ...
+                                       + sym(full(trans.bVect(j,1)))) ...
+                                      ^sym(full(pp)))';
+                      else
                         cc = sym2poly((trans.Amat(j,j)*y+trans.bVect(j,1))^pp)';
+                      end
                         supSet = ineqPolySys0{k}.supports(i*ones(pp+1,1),:);
                         supSet(:,j) = (pp:-1:0)';
 %                
