--- DesignOfIIRFilters/src/deczky3a_sqp_test.m	2020-04-25 10:56:00.178627032 +1000
+++ DesignOfIIRFilters.VM/src/deczky3a_sqp_test.m	2020-04-25 10:55:33.172864854 +1000
@@ -18,8 +18,8 @@
 
 % Deczky3 Lowpass filter specification
 U=0,V=0,Q=6,M=10,R=1
-fap=0.15,dBap=0.1,Wap=1
-fas=0.3,dBas=50,Was=4
+fap=0.15,dBap=0.2,Wap=1
+fas=0.3,dBas=50,Was=2
 ftp=0.25,tp=9.35,tpr=0.3,Wtp=0.0001
 
 % Initial coefficients
--- DesignOfIIRFilters/src/iir_frm_socp_slb_test.m	2020-04-25 10:56:00.183626988 +1000
+++ DesignOfIIRFilters.VM/src/iir_frm_socp_slb_test.m	2020-04-25 10:55:33.245864212 +1000
@@ -52,7 +52,7 @@
 %
 n=800;
 tol=1e-3
-ctol=tol/200
+ctol=tol/100
 maxiter=5000
 verbose=false
 Mmodel=9 % Model filter decimation
--- DesignOfIIRFilters/src/iir_sqp_slb_bandpass_test.m	2020-04-25 10:56:00.184626979 +1000
+++ DesignOfIIRFilters.VM/src/iir_sqp_slb_bandpass_test.m	2020-04-25 10:55:33.256864115 +1000
@@ -13,7 +13,7 @@
 verbose=false
 tol=5e-4
 ctol=tol/100
-maxiter=5000
+maxiter=10000
 
 % Bandpass filter specification
 % (frequencies are normalised to sample rate)
@@ -53,7 +53,7 @@
 close
 
 % Coefficient constraints
-dmax=0.05;
+dmax=0.02;
 [xl,xu]=xConstraints(U,V,M,Q);
 
 % Amplitude constraints
--- DesignOfIIRFilters/src/iir_sqp_slb_fir_bandpass_test.m	2020-04-25 10:56:00.184626979 +1000
+++ DesignOfIIRFilters.VM/src/iir_sqp_slb_fir_bandpass_test.m	2020-04-25 10:55:33.258864097 +1000
@@ -19,7 +19,7 @@
 % Bandpass filter specification
 % (frequencies are normalised to sample rate)
 fapl=0.1,fapu=0.2,dBap=1,Wap=1
-fasl=0.05,fasu=0.25,dBas=36,Wasl=8,Wasu=2
+fasl=0.05,fasu=0.25,dBas=35,Wasl=1,Wasu=1
 
 % Initialise strings
 strM=sprintf(
--- DesignOfIIRFilters/src/iir_sqp_slb_minimum_phase_test.m	2020-04-25 10:56:00.184626979 +1000
+++ DesignOfIIRFilters.VM/src/iir_sqp_slb_minimum_phase_test.m	2020-04-25 10:55:33.261864071 +1000
@@ -18,11 +18,11 @@
 dBap=0.2
 Wap=1
 fas=0.2
-dBas=46
-Was=5
+dBas=40
+Was=1
 
 % Frequency vectors
-n=1000;
+n=800;
 
 % Amplitude constraints
 nap=ceil(n*fap/0.5)+1;
--- DesignOfIIRFilters/src/mcclellanFIRdifferentiator_test.m	2020-04-25 10:56:00.186626961 +1000
+++ DesignOfIIRFilters.VM/src/mcclellanFIRdifferentiator_test.m	2020-04-25 10:55:33.289863824 +1000
@@ -28,7 +28,7 @@
 %
 
 % Specification: low pass filter order is 2*M, length is 2*M+1
-M=500;fap=0.246;fas=0.25;Kp=0.75;Kt=10;
+M=200;fap=0.246;fas=0.25;Kp=0.75;Kt=10;
 
 % Constants 
 f=(0:nf)'*0.5/nf;
--- DesignOfIIRFilters/src/parallel_allpass_delay_sqp_slb_test.m	2020-04-25 10:56:00.188626944 +1000
+++ DesignOfIIRFilters.VM/src/parallel_allpass_delay_sqp_slb_test.m	2020-04-25 10:55:33.309863648 +1000
@@ -24,7 +24,7 @@
 dBap=0.04
 Wap=1
 fas=0.2
-dBas=43
+dBas=40
 Was=50
 
 % Initial coefficients found by tarczynski_parallel_allpass_delay_test.m
--- DesignOfIIRFilters/src/parallel_allpass_socp_slb_test.m	2020-04-25 10:56:00.188626944 +1000
+++ DesignOfIIRFilters.VM/src/parallel_allpass_socp_slb_test.m	2020-04-25 10:55:33.319863560 +1000
@@ -27,7 +27,7 @@
 del.dtol=tol;
 del.stol=ctol;
 warning("Using coef. delta tolerance=%g, SeDuMi eps=%g\n",del.dtol,del.stol);
-n=2000;
+n=1000;
 polyphase=false
 difference=false
 rho=0.999
@@ -39,17 +39,17 @@
 mb=length(Db0)-1
 fape=0.14
 fap=0.15
-dBap=0.02
+dBap=0.03
 Wap=1
 Wape=Wap/4 % Extra passband weight increasing linearly from fape to fap
 ftp=0
 td=0
 tdr=0
 Wtp=0
-fas=0.17075
+fas=0.17
 fase=0.18
-dBas=84.02
-Was_mmse=5e5
+dBas=84
+Was_mmse=1e5
 Wase=Was_mmse/4 % Extra passband weight decreasing linearly from fas to fase
 Was_pcls=1e-6 % Do not attempt to minimise MMSE squared error
 
--- DesignOfIIRFilters/src/polyphase_allpass_socp_slb_test.m	2020-04-25 10:56:00.189626935 +1000
+++ DesignOfIIRFilters.VM/src/polyphase_allpass_socp_slb_test.m	2020-04-25 10:55:33.327863489 +1000
@@ -25,11 +25,11 @@
 % Lowpass filter specification for polyphase combination of all-pass filters
 tol=1e-4
 ctol=1e-9
-n=500;
+n=400;
 polyphase=true
 difference=false
 rho=0.999
-K=1e2
+K=50
 Ksq=K^2;
 R=2
 Ra=R
@@ -37,7 +37,7 @@
 ma=length(Da0)-1
 mb=length(Db0)-1
 fap=0.24
-dBap=0.001
+dBap=0.01
 Wap=1
 fas=0.26
 dBas=80
--- DesignOfIIRFilters/src/schurOneMlattice_socp_slb_bandpass_test.m	2020-04-25 10:56:00.195626882 +1000
+++ DesignOfIIRFilters.VM/src/schurOneMlattice_socp_slb_bandpass_test.m	2020-04-25 10:55:33.428862600 +1000
@@ -11,14 +11,14 @@
 
 
 tol_mmse=2e-5
-tol_pcls=1e-5
+tol_pcls=2e-5
 ctol=tol_pcls
 maxiter=2000
 verbose=false
 
 % Bandpass R=2 filter specification
 fapl=0.1,fapu=0.2,dBap=2,Wap=1
-fasl=0.05,fasu=0.25,dBas=36
+fasl=0.05,fasu=0.25,dBas=33
 Wasl_mmse=5e5,Wasu_mmse=5e5,Wasl_pcls=5e5,Wasu_pcls=5e5
 ftpl=0.09,ftpu=0.21,tp=16,tpr=tp/400,Wtp_mmse=5,Wtp_pcls=5
 
--- DesignOfIIRFilters/src/schurOneMPAlattice_socp_slb_lowpass_to_multiband_test.m	2020-04-25 10:56:00.197626864 +1000
+++ DesignOfIIRFilters.VM/src/schurOneMPAlattice_socp_slb_lowpass_to_multiband_test.m	2020-04-25 10:55:33.449862415 +1000
@@ -12,7 +12,7 @@
 tol=1e-4;
 mtol=1e-4;
 ptol=1e-4;
-ctol=1e-6;
+ctol=1e-5;
 maxiter=2000;
 verbose=false;
 nplot=1000;
--- DesignOfIIRFilters/src/sdp_relaxation_schurOneMlattice_bandpass_10_nbits_test.m	2020-04-25 10:56:00.197626864 +1000
+++ DesignOfIIRFilters.VM/src/sdp_relaxation_schurOneMlattice_bandpass_10_nbits_test.m	2020-04-25 10:55:33.459862327 +1000
@@ -12,16 +12,17 @@
 
 tic;
 
-maxiter=2000
+maxiter=10000
 verbose=false;
-tol=2e-5;
+tol=1e-4;
 ctol=tol;
+
 strf="sdp_relaxation_schurOneMlattice_bandpass_10_nbits_test";
 
 % Bandpass R=2 filter specification
 fapl=0.1,fapu=0.2,dBap=2,Wap=1
 fasl=0.05,fasu=0.25,dBas=33,Wasl=5e5,Wasu=1e6
-ftpl=0.09,ftpu=0.21,tp=16,tpr=0.16,Wtp=5
+ftpl=0.09,ftpu=0.21,tp=16,tpr=0.2,Wtp=1
 
 % Initial filter (found by schurOneMlattice_sqp_slb_bandpass_test.m)
 k0 = [   0.0000000000,   0.6627692632,   0.0000000000,   0.4986334640, ... 
--- DesignOfIIRFilters/src/socp_relaxation_schurOneMPAlattice_bandpass_12_nbits_test.m	2020-04-25 10:56:00.198626855 +1000
+++ DesignOfIIRFilters.VM/src/socp_relaxation_schurOneMPAlattice_bandpass_12_nbits_test.m	2020-04-25 10:55:33.484862107 +1000
@@ -74,8 +74,8 @@
 ftpl=0.09
 ftpu=0.21
 td=16
-tdr=0.2
-Wtp=2
+tdr=0.4
+Wtp=5
 
 %
 % Frequency vectors
